crates/turborepo-lib/src/shim.rs

```diff
@@ -176,50 +177,237 @@ struct PackageJson {
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct YarnRc {
+    pnp_unplugged_folder: PathBuf,
 }
 
-impl LocalTurboState {
-    pub fn infer(repo_root: &Path) -> Option<Self> {
-        let local_turbo_path = repo_root.join("node_modules").join(".bin").join({
+impl Default for YarnRc {
+    fn default() -> Self {
+        Self {
+            pnp_unplugged_folder: PathBuf::from("").join(".yarn").join("unplugged"),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct TurboState {
+    bin_path: Option<PathBuf>,
+    version: &'static str,
+    repo_state: Option<RepoState>,
+}
+
+impl Default for TurboState {
+    fn default() -> Self {
+        Self {
+            bin_path: env::current_exe().ok(),
+            version: get_version(),
+            repo_state: None,
+        }
+    }
+}
+
+impl TurboState {
+    pub fn platform_package_name() -> String {
+        let arch: &'static str = {
+            #[cfg(target_arch = "x86_64")]
+            {
+                "64"
+            }
+            #[cfg(target_arch = "aarch64")]
+            {
+                "arm64"
+            }
+            #[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))]
+            {
+                "unknown"
+            }
+        };
+
+        let os: &'static str = {
+            #[cfg(target_os = "macos")]
+            {
+                "darwin"
+            }
+            #[cfg(target_os = "windows")]
+            {
+                "windows"
+            }
+            #[cfg(target_os = "linux")]
+            {
+                "linux"
+            }
+            #[cfg(not(any(target_os = "macos", target_os = "windows", target_os = "linux")))]
+            {
+                "unknown"
+            }
+        };
+
+        format!("turbo-{}-{}", os, arch)
+    }
+
+    pub fn binary_name() -> &'static str {
+        {
             #[cfg(windows)]
             {
-                "turbo.cmd"
+                "turbo.exe"
             }
             #[cfg(not(windows))]
             {
                 "turbo"
             }
-        });
-
-        if !local_turbo_path.exists() {
-            debug!(
-                "No local turbo binary found at: {}",
-                local_turbo_path.display()
-            );
-            return None;
         }
+    }
+
+    #[allow(dead_code)]
+    pub fn version() -> &'static str {
+        include_str!("../../../version.txt")
+            .lines()
+            .next()
+            .expect("Failed to read version from version.txt")
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct LocalTurboState {
+    bin_path: PathBuf,
+    version: String,
+}
+
+impl LocalTurboState {
+    // Hoisted strategy:
+    // - `npm install`
+    // - `yarn`
+    // - `yarn install --flat`
+    // - berry (nodeLinker: "node-modules")
+    //
+    // This also supports people directly depending upon the platform version.
```

Comment:
Not sure I understand what this means

Reply:
It's technically possible to depend directly on `turbo-darwin-arm64` without using `turbo` for indirection; though grabbing the binary is a bit more obnoxious. This sidesteps that entirely.